### PR TITLE
Allow use of K8s annotations in discovery rules

### PR DESCRIPTION
**Pull request by keitwb**
Had to rework the Get helper function used by govaulate in the rules to
make it work with any map type as well.

### CODE REVIEW

File: test-services/minikube/start-minikube.sh [hunk 1]
```diff
@@ -45,9 +45,6 @@ MINIKUBE_OPTIONS="--vm-driver=none --bootstrapper=${BOOTSTRAPPER} --kubernetes-v
 KUBEADM_OPTIONS="--feature-gates=CoreDNS=true \
     --extra-config=kubelet.authorization-mode=AlwaysAllow \
     --extra-config=kubelet.anonymous-auth=true"
-if [[ "$K8S_VERSION" =~ ^v1\.11\. ]]; then
```
Comment (keitwb) [inline test-services/minikube/start-minikube.sh]: @jchengsfx do you remember why this was there?  We shouldn't really be using the cadvisor port for stats anymore anyway.

Comment (jchengsfx) [reply test-services/minikube/start-minikube.sh]: In minikube, cadvisor was enabled by default for k8s versions <1.11, disabled by default in 1.11, and then removed in 1.12.  We have a cadvisor test for it (now skipped if testing in k8s >=1.12). If this is now unnecessary, we can delete this and disable/remove the test.

File: Makefile [hunk 2]
```diff
@@ -243,9 +243,42 @@ run-chef-tests:
...
+
+	echo "Minikube is now running. Push up an agent image to minikube-test:5000/signalfx-agent:latest or run 'make push-minikube-agent'"
+
+.PHONY: setup-minikube-host
+.SILENT:
+setup-minikube-host:
+	sudo perl -ni -e 'print unless /minikube-test/' /etc/hosts
+	sudo bash -c "echo '$(shell docker inspect --format '{{.NetworkSettings.IPAddress}}' minikube) minikube-test' >> /etc/hosts"
+	sudo mkdir -p /etc/docker/certs.d/minikube-test\:5000/
+	sudo cp test-services/minikube/minikube-test.crt /etc/docker/certs.d/minikube-test\:5000/ca.crt
```
Comment (jchengsfx) [inline Makefile]: Need to include `sudo` package in dev-image.

File: Makefile [hunk 3]
```diff
@@ -243,9 +243,42 @@ run-chef-tests:
...
+		--build-arg 'MINIKUBE_VERSION=$(MINIKUBE_VERSION)' \
+		test-services/minikube
+
+	docker rm -fv minikube 2>/dev/null || true
+	docker run -d \
+		--name minikube \
+		--privileged \
+		-e K8S_VERSION=$(K8S_VERSION) \
+		-e TIMEOUT=$(MINIKUBE_TIMEOUT) \
+		minikube:$(MINIKUBE_VERSION)
```
Comment (jchengsfx) [inline Makefile]: Need to bind port 5000:5000 for the minikube registry?

File: tests/conftest.py [hunk 4]
```diff
@@ -86,111 +97,36 @@ def pytest_generate_tests(metafunc):
...
+        ), f"You must start the minikube container ({minikube_container_name}) before running pytest by running `make minikube`."
+
+        # Pull the kubeconfig out of the minikube container and make a cluster instance based on it.
+        minikube_cont = dclient.containers.get(minikube_container_name)
+        _, kubeconfig_bytes = minikube_cont.exec_run("cat /kubeconfig")
+
+        _, kube_config_path = tempfile.mkstemp()
+        request.addfinalizer(lambda: os.remove(kube_config_path))
+
+        with open(kube_config_path, "wb") as fd:
```
Comment (jaysfx) [inline tests/conftest.py]: `Path("path").write_bytes(kubeconfig_bytes)` is a nice one-liner for this. It also has `read_text()` for reading and `write_text()` for writing.

File: tests/helpers/kubernetes/cluster.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                for pod in utils.get_all_pods(namespace):
+                    print(self.exec_kubectl(f"describe pod {pod.metadata.name}", namespace=namespace))
+
+    @contextmanager
+    def create_resources(self, yamls, timeout=utils.K8S_CREATE_TIMEOUT):
+        resources = []
+        for yaml_file in yamls:
+            assert os.path.isfile(yaml_file), '"%s" not found!' % yaml_file
+            with open(yaml_file, "r") as fd:
+                for doc in yaml.load_all(fd.read()):
```
Comment (jchengsfx) [inline tests/helpers/kubernetes/cluster.py]: Should use `yaml.safe_load_all()`.

**Pull request merged**